src/hotspot/share/opto/type.cpp

```diff
@@ -658,6 +661,11 @@ void Type::Initialize_shared(Compile* current) {
   // get_zero_type() should not happen for T_CONFLICT
   _zero_type[T_CONFLICT]= NULL;
 
+#if defined(X86)
```

Comment:
Yes, SVE should also be fine with this. Perhaps to use the default type TypeInt::BYTE instead of TypeInt::BOOL? It seems that masking byte type looks more reasonable?